aten/src/ATen/native/cuda/ScanKernels.cu

```diff
@@ -225,4 +225,205 @@ void cummin_helper_cuda(const Tensor& self, Tensor& values, Tensor& indices, int
   });
 }
 
+/* Perform an inclusive scan along an outer dimension of a tensor.
```

Comment:
Could you add a note here also that explains that this implementation is very close to the other scan above in this file and should be merged together to reduce duplicate code, but this is left as a future PR.